### PR TITLE
Add attribute for fileprotection

### PR DESCRIPTION
**Pull request by cdhoffmann**
Adds an attribute for our FileSystem storage directory and sets the fileprotection to none. This is to avoid system restrictions during app backgrounding and other potential edge cases.

Also adds capabilities to our test app for testing backgrounding tasks.

### CODE REVIEW

File: AEPServices/Sources/storage/FileSystemNamedCollection.swift [hunk 1]
```diff
@@ -126,6 +126,7 @@ class FileSystemNamedCollection: NamedCollectionProcessing {
             let adobeBaseUrl = baseUrl.appendingPathComponent(adobeDirectory, isDirectory: true)
             do {
                 try fileManager.createDirectory(at: adobeBaseUrl, withIntermediateDirectories: true)
+                try fileManager.setAttributes([FileAttributeKey.protectionKey: FileProtectionType.none], ofItemAtPath: adobeBaseUrl.path)
             } catch {
                 Log.error(label: adobeDirectory, "Failed to create storage directory with error: \(error)")
```
Comment (praveek) [inline AEPServices/Sources/storage/FileSystemNamedCollection.swift:131]: Should we consider adjusting this error message? Is it possible to check the reason and avoid printing this message when the error is a result of the directory already existing?

Comment (cdhoffmann) [reply AEPServices/Sources/storage/FileSystemNamedCollection.swift:131]: I believe it doesn't error when the directory already exists

Comment (cdhoffmann) [reply AEPServices/Sources/storage/FileSystemNamedCollection.swift:131]: https://github.com/adobe/aepsdk-core-ios/pull/966#discussion_r1318920171

**Review by praveek — APPROVED** [review]
